source/agora/consensus/protocol/Nominator.d

```diff
@@ -529,8 +529,14 @@ extern(D):
             const Block prev_block = blocks.front;
             Hash random_seed = this.ledger.getExternalizedRandomSeed(
                 this.ledger.getBlockHeight(), con_data.missing_validators);
-            const Block proposed_block = makeNewBlock(prev_block, con_data.tx_set,
-                con_data.timestamp, con_data.enrolls, random_seed, con_data.missing_validators);
+
+            Transaction[] received_tx_set;
+            if (auto fail_reason = this.ledger.getValidTXSet(con_data, received_tx_set))
+                return; // We dont have all the TXs for this block. Try to catchup
```

Comment:
We probably should log here the fail_reason as there is nothing logged in any of the called functions.